ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,408 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+#####################################
+##############Initialize#############
+#####################################
+
+workdir=`mktemp -d`
+
+bastion_ignition_file="${workdir}/bastion.ign"
+ssh_pub_keys_file="/var/run/vault/qe-ssh-key/public_key"
+reg_cert_file="/var/run/vault/qe-mirror-registry/server_domain.crt"
+reg_key_file="/var/run/vault/qe-mirror-registry/server_domain.pem"
+src_proxy_creds_file="/var/run/vault/qe-proxy/proxy_creds"
+src_proxy_creds_encrypted_file="/var/run/vault/qe-proxy/proxy_creds_encrypted_apr1"
+src_registry_creds_encrypted_file="/var/run/vault/qe-mirror-registry/registry_creds_encrypted_htpasswd"
```

Comment:
I think that need permission to update the CLUSTER_PROFILE vault collections, right?

Reply:
Yeah, because we have no permission to update all the existing CLUSTER_PROFILE vault collections, and to ensure QE always can access these services on our own, maybe this approach is more easier to manage.

Reply:
Okay, so if the cluster profile is using `azure-4`, we can define a collections in our vault named `azure-4`, include our desired key-value there, then our defined items will override the original ones?

Reply:
yes, you can do that, only one thing need to be careful, if there is a conflict between the two keys/names, the one in vault will override the existing one.